homeassistant/components/ads/valve.py

```diff
@@ -45,30 +45,27 @@ def setup_platform(
     ads_var: str = config[CONF_ADS_VAR]
     name: str = config[CONF_NAME]
     device_class: ValveDeviceClass | None = config.get(CONF_DEVICE_CLASS)
-    supported_features: ValveEntityFeature = (
-        ValveEntityFeature.OPEN | ValveEntityFeature.CLOSE
-    )
 
-    entity = AdsValve(ads_hub, ads_var, name, device_class, supported_features)
+    entity = AdsValve(ads_hub, ads_var, name, device_class)
 
     add_entities([entity])
 
 
 class AdsValve(AdsEntity, ValveEntity):
     """Representation of an ADS valve entity."""
 
+    _attr_supported_features = ValveEntityFeature.OPEN | ValveEntityFeature.CLOSE
+
```

Comment:
This is the only code change => making the attribute static
All other changes are type-hint only.